llvm/lib/ObjCopy/MachO/MachOLayoutBuilder.cpp

```diff
@@ -173,7 +173,7 @@ uint64_t MachOLayoutBuilder::layoutSegments() {
         if (!Sec->hasValidOffset()) {
           Sec->Offset = 0;
         } else {
-          if (RequiresFirstSectionOutsideFirstPage) {
+          if (RequiresFirstSectionOutsideFirstPage && SectOffset < PageSize) {
```

Comment:
Wouldn't this cause issues w/ the sections that come after?  If offset for a given section is updated, it may overlap some other.

Reply:
Approving, as per our chat during the call, it should never happen for our particular case.  Agreed :+1:  that the general case is more involved; up to the maintainers (or maybe us in the future) to write a proper fix.